tinygrad/runtime/ops_torch.py

```diff
@@ -18,7 +18,7 @@ def as_strided(x, arg):
 torch_fxn_for_op: Dict[Op, Callable] = {**base_fxn_for_op, **{
   UnaryOps.NOOP: lambda x: x.contiguous(), UnaryOps.SQRT: lambda x: x.sqrt(), UnaryOps.EXP2: lambda x: x.exp2(), UnaryOps.LOG2: lambda x: x.log2(), UnaryOps.SIN: torch.sin,
   UnaryOps.CAST: lambda x,y: (x.view if y[1] else x.type)(next(k for k,v in type_map.items() if v==y[0])),
-  BinaryOps.MAX: torch.maximum, BinaryOps.CMPLT: lambda x,y: (x<y).type(torch.promote_types(x.dtype, y.dtype)),
+  BinaryOps.MAX: torch.maximum, BinaryOps.CMPLT: lambda x,y: (x<y).type(torch.promote_types(x.dtype, y.dtype)), BinaryOps.SUB: lambda x,y: torch.logical_xor(x, y) if x.dtype == torch.bool or y.dtype == torch.bool else torch.sub(x, y),
```

Comment:
It could be 
```
BinaryOps.SUB: lambda x,y: torch.logical_xor(x, y) if y.dtype is torch.bool else torch.sub(x, y),
```
This would cover for the case of the `__ge__`, `__le__`, and `__eq__`, where self and x are both `dtype.bool`
but however
```
(Tensor(0, dtype=dtypes.bool, device="TORCH") - Tensor(1, dtype=dtypes.int32, device="TORCH")).realize()
```
this would still return https://github.com/tinygrad/tinygrad/issues/1524 this error 
though I don't think this edge case is right anyways...
Why would anyone sub a bool with an int...